tfc-dump.pl

```diff
@@ -115,29 +115,39 @@
 my @vs_names;
 my @vs_ids;
 my $pg_size = 100;
+my $pg_num = 1;
 my $total_count;
 my $tmpfile = `mktemp`;
 chomp($tmpfile);
 
-$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/varsets?page%5Bsize%5D=${pg_size}\"";
-$curl_cmd   = "curl $curl_headers --output $tmpfile $curl_query";
-system($curl_cmd);
-
-# Get the Variable Set names
-
-$jq_cmd   = "cat $tmpfile | jq '.data[].attributes.name'";
-@vs_names = `$jq_cmd`;
-# Remove the double quotes in each element of the array.
-grep($_ =~ s/"//g && 0, @vs_names);	# Programming Perl, p. 221, 1990
-chomp(@vs_names);
-
-# Get the Variable Set IDs
+do {
+	$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/varsets?page%5Bsize%5D=${pg_size}&page%5number%5D=${pg_num}\"";
+	$curl_cmd   = "curl $curl_headers --output $tmpfile $curl_query";
+	system($curl_cmd);
 
-$jq_cmd = "cat $tmpfile | jq '.data[].id'";
-@vs_ids = `$jq_cmd`;
-# Remove the double quotes in each element of the array.
-grep($_ =~ s/"//g && 0, @vs_ids);	# Programming Perl, p. 221, 1990
-chomp(@vs_ids);
+	# Get the Variable Set names
+	$jq_cmd   = "cat $tmpfile | jq '.data[].attributes.name'";
+	my @page_names = `$jq_cmd`;
+
+	# Remove the double quotes in each element of the array.
+	grep($_ =~ s/"//g && 0, @page_names);	# Programming Perl, p. 221, 1990
```

Comment:
Yuck! I would prefer a more readable version that does not rely on side-effects, even though it is less memory efficient:
```suggestion
	@page_names = map { s/"//gr } @page_names;
```